packages/server/graphql/public/mutations/startRecurrence.ts

```diff
@@ -37,26 +40,76 @@ const startRecurrence: MutationResolvers['startRecurrence'] = async (
     return standardError(new Error('Meeting is not a team prompt meeting'), {userId: viewerId})
   }
 
-  if (meeting.meetingSeriesId) {
-    return standardError(new Error('Meeting already has meeting series'), {userId: viewerId})
-  }
-
   // Next meeting start is tomorrow at 9a UTC
   // :TODO: (jmtaber129): Determine this from meeting series configuration.
-  const startDate = new Date(
+  const nextMeetingStartDate = new Date(
     Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate() + 1, 9)
   )
   const recurrenceRule = new RRule({
     freq: RRule.WEEKLY,
     byweekday: [RRule.MO, RRule.TU, RRule.WE, RRule.TH, RRule.FR],
-    dtstart: startDate
+    dtstart: nextMeetingStartDate
   })
 
+  if (meeting.meetingSeriesId) {
+    const {cancelledAt, recurrenceRule, id} = await dataLoader
+      .get('meetingSeries')
+      .loadNonNull(meeting.meetingSeriesId)
+    if (!cancelledAt) {
+      return standardError(new Error('Meeting is already recurring'), {userId: viewerId})
+    }
+
+    const currentRRule = RRule.fromString(recurrenceRule)
+    const hasRecentlyStopped = cancelledAt.getTime() > now.getTime() - ms('1h')
+    const nextScheduledStartDate = currentRRule.after(now)
+    const isNextScheduledStartDateReasonable =
+      nextScheduledStartDate.getTime() > now.getTime() + ms('9h')
+    if (hasRecentlyStopped || isNextScheduledStartDateReasonable) {
+      // no need to change the recurrence rule, next meeting starts in reasonable time
+      await restartMeetingSeries({cancelledAt: null}, meeting.meetingSeriesId)
+    } else {
+      // meeting series was stopped more than ~1h ago and the next scheduled start date is within the next ~9h
+      // to prevent a meeting from being scheduled too soon, we need to change the recurrence rule
+      const newRecurrenceRule = currentRRule.clone()
+      newRecurrenceRule.options.dtstart = new Date(nextMeetingStartDate.getTime() + ms('1d'))
+
+      await restartMeetingSeries(
+        {cancelledAt: null, recurrenceRule: newRecurrenceRule.toString()},
+        meeting.meetingSeriesId
+      )
+    }
+
+    dataLoader.get('meetingSeries').clear(meeting.meetingSeriesId)
+
+    // update all the active meetings to end at proper time
+    const activeMeetings = await r
+      .table('NewMeeting')
+      .getAll(id, {index: 'meetingSeriesId'})
+      .filter({endedAt: null}, {default: true})
+      .run()
+    const updates = activeMeetings.map((meeting) =>
+      r
+        .table('NewMeeting')
+        .get(meeting.id)
+        .update({
+          scheduledEndTime: new Date(
+            meeting.createdAt.getTime() + ms(`${MEETING_DURATION_IN_MINUTES}m`)
+          )
+        })
+        .run()
+    )
+    await Promise.all(updates)
+
```

Comment:
@tianrunhe do we want to send an event about restarting recurrence?

Reply:
Thanks for the mention! Yeah I think we need to send an event here but to start, we could just use the existing `Meeting Recurrence Started` event. That being said, can you also add `Meeting Recurrence Ended` in `stopRecurrence` mutation? My thinking is that with these two, we could dive deep into analysis of restarting behavior if we want to (because we can join them by `meetingSeriesId`) so no need to send a new `Meeting Recurrence Restarted` event. WDYT?

Reply:
sounds good to me! will do.

Reply:
only issue I see right now is the fact that we send meeting properties along with the `Meeting Recurrence Started`, like that: 
```
this.track(userId, 'Meeting Recurrence Started', createMeetingProperties(meeting))
```

It's technically possible to have many meetings opened a the same time related to the same meeting series, IMO we should send meeting series params, instead of meeting. For example, what meeting id should we send in the case below? IMO, it'd be nice to have that sorted out, before recurrence goes to prod and I'd suggest to send meeting series params there. 

```

    // update all the active meetings to end at proper time
    const activeMeetings = await r
      .table('NewMeeting')
      .getAll(id, {index: 'meetingSeriesId'})
      .filter({endedAt: null}, {default: true})
      .run()
    const updates = activeMeetings.map((meeting) =>
      r
        .table('NewMeeting')
        .get(meeting.id)
        .update({
          scheduledEndTime: new Date(
            meeting.createdAt.getTime() + ms(`${MEETING_DURATION_IN_MINUTES}m`)
          )
        })
        .run()
    )
    await Promise.all(updates)

    analytics.recurrenceStarted(viewerId, ??)
```

what do you think @tianrunhe @jmtaber129 ?

Reply:
Yeah good tension! I think initially James [just re-used the helper function](https://github.com/ParabolInc/parabol/pull/7041#discussion_r947443676) so we can easily get common properties between `Meeting` and `MeetingSeries` but I agree, in this case, we only want **1** `Meeting Series Started` event. In that event I don't really need to know the `meetingId`? As A) there are more than one, B) if I want to know which meetings are affected, I can join by `meetingSeriesId`.
Yeah so it feels like we should have treat `MeetingSeries` separately.

Reply:
@tianrunhe @BartoszJarocki IMO, since all recurrence-related mutation are triggered from a specific meeting, the user action here is "**restart recurrence for this meeting**" (even though the effect on the data model is "restart recurrence for this _series_"), so I think it makes sense to continue associating this event with the meeting passed to the mutation (at least for now).

Re: reusing the "Meeting Recurrence Started" event - unless there's an easy way to make a custom event in Amplitude on "first event for a distinct property value", we should track a flag to differentiate initial recurrence starts and recurrence restarts for analysis within Amplitude. E.g. I might want to investigate questions like "what % of recurrence starts are immediately cancelled, broken down by whether it's an initial start or restart".

Later on, we'll probably change our user-facing semantics to be closer to "restart recurrence for meeting _series_", but we can cross that bridge re: analytics when we get to it.

Reply:
@jmtaber129 I understand the arguments, but IMO it's getting harder and harder to figure out what `meetingId` to pass in certain cases. I'd like to have this PR merged as is and let's create a separate task for analytics if needed.